openmetadata-service/src/main/java/org/openmetadata/service/migration/api/MigrationStep.java

```diff
@@ -0,0 +1,34 @@
+package org.openmetadata.service.migration.api;
+
+import org.jdbi.v3.core.Handle;
+import org.openmetadata.service.jdbi3.locator.ConnectionType;
+
+public interface MigrationStep {
+  // This version should match the server version
+  // Ex: if the server is 1.0.0 the migration version for that server is 1.0.0
+  // This version is used to sort all the upgrade migrations and apply them in order
+  double getMigrationVersion();
+
+  String getMigrationFileName();
+
+  ConnectionType getDatabaseConnectionType();
+
+  void initialize(Handle handle);
+
+  // Handle Non-transactional supported SQLs here Example changes in table struct (DDL
+  void nonTransactionalPreDataMigrationSQL();
```

Comment:
this should be part of runDataMigration right. I think adding more methods will be confusing for anyone new on where to add. That's why started with a simple structure